### PR TITLE
SmartSearchControl draft

### PR DESCRIPTION
**Pull request by codyebberson**

### CODE REVIEW

File: packages/app/src/SmartSearchPage.tsx [hunk 1]
```diff
@@ -0,0 +1,108 @@
+import { PropertyType } from '@medplum/core';
+import { MemoizedSmartSearchControl, SmartSearchField } from '@medplum/ui';
+import React from 'react';
+import { useLocation, useNavigate } from 'react-router-dom';
+
+const gql = `{
```
Comment (codyebberson) [inline packages/app/src/SmartSearchPage.tsx:6]: This would come from a query param

File: packages/app/src/SmartSearchPage.tsx [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+      high {
+        value,
+        unit
+      }
+    }
+  }
+}
+}`;
+
+const fields: SmartSearchField[] = [
```
Comment (codyebberson) [inline packages/app/src/SmartSearchPage.tsx:51]: This would come from a query param

File: packages/app/src/SmartSearchPage.tsx [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+    name: 'TESTO High',
+    fhirPath: 'ObservationList.referenceRange.high',
+    propertyType: PropertyType.Quantity,
+  },
+];
+
+export function SmartSearchPage(): JSX.Element {
+  const navigate = useNavigate();
+  const location = useLocation();
+  const resourceType = 'ServiceRequest';
```
Comment (codyebberson) [inline packages/app/src/SmartSearchPage.tsx:92]: This would come from a query param

File: packages/ui/src/SmartSearchControl.tsx [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+
+export interface SmartSearchResponse {
+  data: {
+    ResourceList: Resource[];
+  };
+}
+
+/**
+ * The SmartSearchControl component represents the embeddable search table control.
+ * It includes the table, rows, headers, sorting, etc.
```
Comment (codyebberson) [inline packages/ui/src/SmartSearchControl.tsx:37]: This comment is wrong.  Copy pasta

**Review by reshmakh — APPROVED** [review]

**Review by rahul1 — APPROVED** [review]

**Pull request merged**